### PR TITLE
add pyproject.toml and black, updated reqs

### PR DESCRIPTION
**Pull request by xiahongze**
bump version for some of the packages
add black
add pyproject.toml

### CODE REVIEW

File: {{cookiecutter.project_slug}}/.vscode/settings.json [hunk 1]
```diff
@@ -1,10 +1,11 @@
 {
     "python.linting.enabled": true,
     "python.linting.pylintPath": "pylint",
-    "python.formatting.provider": "yapf",
```
Comment (StephenHogg) [inline {{cookiecutter.project_slug}}/.vscode/settings.json:4]: I think we actually need to edit the cookiecutter to prompt the user to choose between yapf and black. Ben H's team use yapf at the moment, while the Optimus guys use black. This means that we would want to use black while others using this cookiecutter might want to use yapf. Seem reasonable?

File: {{cookiecutter.project_slug}}/pylintrc [hunk 2]
```diff
@@ -60,7 +60,13 @@ confidence=
 # --enable=similarities". If you want to run only the classes checker, but have
 # no Warning level messages displayed, use "--disable=all --enable=classes
 # --disable=W".
-disable=print-statement,
+disable=redefined-outer-name,
+        invalid-name,
+        missing-class-docstring,
```
Comment (StephenHogg) [inline {{cookiecutter.project_slug}}/pylintrc:65]: Any reason we're ignoring the absence of docstrings?

File: {{cookiecutter.project_slug}}/.vscode/settings.json (hunk 1 above)
Comment (StephenHogg) [reply {{cookiecutter.project_slug}}/.vscode/settings.json:4]: Is it worth also updating the build scripts in the `scripts` directory to use black/yapf and isort so they become part of the build process always?

File: {{cookiecutter.project_slug}}/setup.py [hunk 3]
```diff
@@ -4,23 +4,26 @@
...
 ]
 
 setup_requirements = [
     'pytest-runner',
 ]
 
 test_requirements = [
-    'pytest>=5.1.2',
-    'pytest-cov>=2.7.1',
+    'pytest>=6.2.0',
```
Comment (StephenHogg) [inline {{cookiecutter.project_slug}}/setup.py:25]: 6.2.0 here but 6.2.1 in requirements.txt

**Review by StephenHogg — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_slug}}/requirements.txt [hunk 4]
```diff
@@ -5,13 +5,13 @@ wheel==0.34.2
...
-pytest==6.2.1
-pytest-cov==2.10.1
+pytest==6.2.0
+pytest-cov==2.10.0
 pytest-runner==5.1
 pytest-xdist==1.32.0
 python-dateutil==2.8.0
 pylint==2.4.1
-black==20.8b1
+{% if cookiecutter.formatter == 'black' %}black==20.8b1{% else %}yapf==0.29.0{% endif %}
```
Comment (StephenHogg) [inline {{cookiecutter.project_slug}}/requirements.txt:14]: add yapf to requirements.txt if it is requested?

**Review by StephenHogg — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_slug}}/requirements.txt (hunk 4 above)
Comment (xiahongze) [reply {{cookiecutter.project_slug}}/requirements.txt:14]: i believe it is added???

File: {{cookiecutter.project_slug}}/pylintrc (hunk 2 above)
Comment (StephenHogg) [reply {{cookiecutter.project_slug}}/pylintrc]: Only this one left, and we're done

Comment (xiahongze) [reply {{cookiecutter.project_slug}}/pylintrc]: please note that there is an outdated icon in this thread, meaning that the code you looked is no longer there.

Comment (StephenHogg) [reply {{cookiecutter.project_slug}}/pylintrc]: Very sorry! I look up to your ability as a coder every day :)

**Review by StephenHogg — APPROVED** [review]

**Comment by StephenHogg** [comment]
PR approved, thanks for this!

**Pull request merged**